nifi-nar-bundles/nifi-adx-bundle/nifi-adx-processors/src/main/java/org/apache/nifi/processors/adx/AzureAdxIngestProcessor.java

```diff
@@ -0,0 +1,437 @@
+            .displayName("IngestionMappingKind")
+            .description("The type of the ingestion mapping related to the table in ADX.")
+            .required(true)
+            .allowableValues(IM_KIND_AVRO, IM_KIND_APACHEAVRO, IM_KIND_CSV, IM_KIND_JSON, IM_KIND_ORC, IM_KIND_PARQUET)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    static final PropertyDescriptor IR_LEVEL = new PropertyDescriptor.Builder()
+            .name("IR_LEVEL")
+            .displayName("IngestionReportLevel")
+            .description("ADX can report events on several levels: None, Failure and Failure&Success.")
+            .required(true)
+            .allowableValues(IRL_NONE, IRL_FAIL, IRL_FAS)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    static final PropertyDescriptor IR_METHOD = new PropertyDescriptor.Builder()
+            .name("IR_METHOD")
+            .displayName("IngestionReportMethod")
+            .description("ADX can report events on several methods: Table, Queue and Table&Queue.")
+            .required(true)
+            .allowableValues(IRM_TABLE, IRM_QUEUE, IRM_TABLEANDQUEUE)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor ADX_SERVICE = new PropertyDescriptor
+            .Builder().name("ADX_SERVICE")
+            .displayName("AzureADXConnectionService")
+            .description("Service that provides the ADX-Connections.")
+            .required(true)
+            .identifiesControllerService(AdxConnectionService.class)
+            .build();
+
+    public static final PropertyDescriptor WAIT_FOR_STATUS = new PropertyDescriptor
+            .Builder().name("WAIT_FOR_STATUS")
+            .displayName("Wait for status")
+            .description("Define the status to be waited on.")
+            .required(true)
+            .allowableValues(ST_SUCCESS, ST_FIREANDFORGET)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final Relationship RL_SUCCEEDED = new Relationship.Builder()
+            .name("RL_SUCCEEDED")
+            .description("Relationship for success")
+            .build();
+
+    public static final Relationship RL_FAILED = new Relationship.Builder()
+            .name("RL_FAILED")
+            .description("Relationship for failure")
+            .build();
+
+    private List<PropertyDescriptor> descriptors;
+
+    private Set<Relationship> relationships;
+
+    private AdxConnectionService service;
+
+    @Override
+    protected void init(final ProcessorInitializationContext context) {
+        final List<PropertyDescriptor> descriptors = new ArrayList<>();
+        descriptors.add(ADX_SERVICE);
+        descriptors.add(DB_NAME);
+        descriptors.add(TABLE_NAME);
+        descriptors.add(MAPPING_NAME);
+        descriptors.add(FLUSH_IMMEDIATE);
```

Comment:
Since we have FLUSH_IMMEDIATE , are we setting this in ingestion properties ?

Reply:
yes we are setting the flush_immediate in the ingestion properties in Line no 387